lldb/unittests/Core/SwiftDemanglingPartsTest.cpp

```diff
@@ -0,0 +1,1263 @@
+//===-- SwiftMangledTest.cpp ----------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "Plugins/LanguageRuntime/Swift/SwiftLanguageRuntime.h"
+#include "TestingSupport/TestUtilities.h"
+
+#include "lldb/Core/DemangledNameInfo.h"
+#include "lldb/Core/Mangled.h"
+
+#include "gtest/gtest.h"
+
+using namespace lldb;
+using namespace lldb_private;
+
+struct SwiftDemanglingPartsTestCase {
+  const char *mangled;
+  DemangledNameInfo expected_info;
+  std::string_view basename;
+  std::string_view arguments;
+};
+
+SwiftDemanglingPartsTestCase g_swift_demangling_parts_test_cases[] = {
+    // clang-format off
+  { "_TFC3foo3bar3basfT3zimCS_3zim_T_",
```

Comment:
Did you copy these from the Swift demangler tests?
If yes, I don't think it's necessary to test all of it again here, we just need to test the interface.
Alternatively, if there is a way to use a clever macro or a generated source file to not duplicate the tests here, that would be most future-proof.

Reply:
A macro could work. How about a macro which reads line `n` of the `manglings.txt` file and generates the appropriate test case:

```cpp
DEMANGLER_TEST_CASE(1, {8, 11}, {0, 0}, {11, 25}, {0, 0}, {0, 8}, {25, 31}, "bas", "(zim: foo.zim)")
```

becomes

```cpp
  { "_TFC3foo3bar3basfT3zimCS_3zim_T_",
    { /*.BasenameRange=*/{8, 11}, /*.ScopeRange=*/{0, 0}, /*.ArgumentsRange=*/{11, 25},
    /*.QualifiersRange=*/{0, 0}, /*.PrefixRange=*/{0, 8}, /*.SuffixRange=*/{25, 31}
    },
    /*.basename=*/"bas",
    /*.arguments=*/"(zim: foo.zim)"
  },
```

Reply:
I don't think having a couple of test-cases here is bad (even if the mangled name was taken from the demangler test-suite). It just tests the basic functionality of the demangler tracking.

But if you want to add a test that will stay up-to-date with latest changes in the demangler, having a common source for test-cases would be the best way to go. I think that can be done as a separate patch though? That's what we do for the ItaniumDemangler for example: https://github.com/llvm/llvm-project/pull/137793

Reply:
> But if you want to add a test that will stay up-to-date with latest changes in the demangler, having a common source for test-cases would be the best way to go. I think that can be done as a separate patch though? That's what we do for the ItaniumDemangler for example: [llvm#137793](https://github.com/llvm/llvm-project/pull/137793)

I agree that this should be done in a separate patch as the existing tests already check if the range tracking works with various edge cases.